dotnet/src/webdriver/Internal/Base64UrlEncoder.cs

```diff
@@ -0,0 +1,159 @@
+// Thanks to Microsoft.IdentityModel.Tokens package for this code.
+// https://raw.githubusercontent.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/6.19.0/src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs
+using System;
+
+namespace OpenQA.Selenium.Internal
+{
+    /// <summary>
+    /// Encodes and Decodes strings as Base64Url encoding.
+    /// </summary>
+    internal static class Base64UrlEncoder
+    {
+        private const char base64PadCharacter = '=';
+        private const string doubleBase64PadCharacter = "==";
+        private const char base64Character62 = '+';
+        private const char base64Character63 = '/';
+        private const char base64UrlCharacter62 = '-';
+        private const char base64UrlCharacter63 = '_';
+
+        /// <summary>
+        /// Encoding table
+        /// </summary>
+        internal static readonly char[] s_base64Table =
+        {
+            'A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z',
+            'a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z',
+            '0','1','2','3','4','5','6','7','8','9',
+            base64UrlCharacter62,
+            base64UrlCharacter63
+        };
+
+        /// <summary>
+        /// Converts a subset of an array of 8-bit unsigned integers to its equivalent string representation which is encoded with base-64-url digits. Parameters specify
+        /// the subset as an offset in the input array, and the number of elements in the array to convert.
+        /// </summary>
+        /// <param name="inArray">An array of 8-bit unsigned integers.</param>
+        /// <param name="length">An offset in inArray.</param>
+        /// <param name="offset">The number of elements of inArray to convert.</param>
+        /// <returns>The string representation in base 64 url encoding of length elements of inArray, starting at position offset.</returns>
+        /// <exception cref="ArgumentNullException">'inArray' is null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">offset or length is negative OR offset plus length is greater than the length of inArray.</exception>
+        public static string Encode(byte[] inArray, int offset, int length)
```

Comment:
Selenium doesn't call this method signature directly, and we end up checking for null twice. Since offset is always 0, we should be able to combine this method into the one below.